### ISSUE TITLE
No field of name "pageInfo" found on type "Product" in schema

### ISSUE THREAD

**Issue by ilbertt** [Shopify/js-buy-sdk#864]
### Bug details 

**Describe the bug**
Trying to get a product with graphQLClient (from `index.unoptimized.umd`) fails.

**To Reproduce**
- write this code in your React app
```javascript
const productQuery = client.graphQLClient.query((root) => {
    root.addConnection('productByHandle', { args: { handle: "my-handle" } }, (product) => {
        product.add('title');
    });
});

client.graphQLClient.send(productQuery).then(({ model, data }) => {
    // Do something with the products
    console.log(model, data);
});
```
- run the app in browser
- get the error:
```
Error in function SelectionSetBuilder.field in ./node_modules/shopify-buy/index.unoptimized.umd.js:873
No field of name "pageInfo" found on type "Product" in schema
```

**Expected behavior**
Function should return the Product element.

**Environment (please complete the following information):**
 - OS: Ubuntu 20.0.4 LTS
 - Browser: Opera
 - SDK Version: ^2.13.0
 
**Bug Report Checklist**
- [x] I have read and agree to the [CODE_OF_CONDUCT.md](https://github.com/Shopify/js-buy-sdk/blob/master/CODE_OF_CONDUCT.md)
- [x] I have read the [CONTRIBUTING.md](https://github.com/Shopify/js-buy-sdk/blob/master/CONTRIBUTING.md) guidelines.
- [x] I have provided a detailed description of the bug, including code samples, and/or network data.
- [x] I have provided information about my development environment, including SDK version.

**Comment by riccardolardi** [comment]
@Luca8991 I worked around it by querying for `products` instead - this seems to work fine:

```javascript
      const metafieldQuery = client.graphQLClient.query((root) => {
        root.addConnection('products', { args: { first: 249 } }, (product) => {
          product.add('handle')
          product.addConnection(
            'metafields',
            { args: { first: 249 } },
            (metafield) => {
              metafield.add('key')
              metafield.add('value')
            }
          )
        })
      })
      const metafieldQueryResult = await client.graphQLClient.send(
        metafieldQuery
      )
```

**Comment by simonhrogers** [comment]
@danieladarve Nope, sorry! Low budget / tight deadline affair so I’ve had to settle for querying all products until someone else can resolve this. As far as I can tell its an issue with the current version of the buy SDK.

**Comment by Kncarrier** [comment]
I have run into this issue too. It looks like this issue is present when trying to query for any singular object and on at least the customer mutations. I am trying the use the customerAccessTokenCreate mutation, but can't get past this error.

`Uncaught Error: No field of name "pageInfo" found on type "CustomerAccessTokenCreatePayload" in schema`

**Comment by liberat0r** [comment]
After a few hours of trial & error (mostly error) here's the latest and greatest version of product metafield fetching.

```
 const productsQuery = this.client.graphQLClient.query((root) => {
	root.addConnection('products', { args: { first: 10 } }, (product) => {
		product.add('title');
		product.add('tags');
		product.add('metafields', {
			args: {
				identifiers: [
					{
						key: "color",
						namespace: "main"
					},
					{
						key: "gender",
						namespace: "main"
					}
				]
			}
		}, (metafield) => {
			metafield.add('value');
		});
	});
});
```

It would be nice to include it as an example on the frontpage to save other people's time.

**Comment by miketu926** [comment]
or we can have additional fields such as metafield/metafields and tags etc to flow through into the optimized version of .fetch and .fetchAll, rather the optimized graphQLClient
